src/vasp.jl

```diff
@@ -167,57 +167,55 @@ readDOSCAR(filename::AbstractString) = open(readDOSCAR, filename)
 
 readDOSCAR() = open(readDOSCAR, "DOSCAR")
 
+"""
+    readPROCAR(io::IO) -> FatBands{3}
+
+Reads an lm-decomposed PROCAR file from VASP and returns its data as a FatBands struct.
+"""
 function readPROCAR(io::IO)
     ln = readline(io)
     
+    # Checks for lm-decomposed PROCAR
     if cmp(split(ln)[2], "lm") == -1
         error("Not a lm-decomposed PROCAR. Use a different PROCAR.")
     end
     
-    # Get number of kpoints, number of bands, number of ions
-    ln = readline(io)
-    nkpt = parse(Int,split(ln)[4])
-    nband = parse(Int,split(ln)[8])
-    nion = parse(Int,split(ln)[12])
+    # Read header for number of kpoints, number of bands, number of ions
+    (nkpt, nband, nion) = parse.(Int, split(readline(io))[[4, 8, 12]])
     
-    # Initializes containers... to be fixed
-    kptlist = KPointList{3}(Vector{SVector{3,Float64}}(undef, nkpt))
-    # Contains kpoint weight. Unused for now.
+    # Initializes containers. Some are unused for now, with potentially useful information is commented out.
+    # kptlist = KPointList{3}(Vector{SVector{3,Float64}}(undef, nkpt))
```

Comment:
Great move to leave the original code in - never hurts to keep it in as a reference in case things change in the future.